### ISSUE TITLE
Why are streamlit_echarts not visualized in multitabs?

### ISSUE THREAD

**Issue by Nemecsek** [andfanilo/streamlit-echarts#43]
Note: published on [Stackoverflow](https://stackoverflow.com/questions/74376137/why-are-streamlit-echarts-not-visualized-in-multitabs)

I want a 4-tab page in Streamlit: 2 echarts + 2 plotly.

**Why is any echart plot (but the first) not visualized, while plotly's ones are ok?**

(The real app needs a Sankey with special features that only Echarts can provide, so using another library (after extensive research) is not feasible: as in this stripped down example, only the first Sankey is visualised, not the next ones, as echarts "forgets" to refresh the visualisation after switching the tab)

```
import streamlit as st
from streamlit_echarts import st_echarts
import plotly.figure_factory as ff
import numpy as np

tab1, tab2, tab3, tab4 = st.tabs(["echarts_tab1", "echarts_tab2", "plotly_tab3", "plotly_tab4"])

def echarts_random_options():
    return {
        "xAxis": {"type": "category", "data": ["Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"]},
        "yAxis": {"type": "value"},
        "series": [
            {"data": list(np.random.random(size=7) * 800),
             "type": "line"}
        ],
    }

def plotly_random_fig():
    return ff.create_distplot([np.random.randn(200)], ["values"], bin_size=[.1, .25, .5])

with tab1:
    st_echarts(options=echarts_random_options())
    
with tab2:
    st_echarts(options=echarts_random_options())
    
with tab3:
    st.plotly_chart(plotly_random_fig(), use_container_width=True)

with tab4:
    st.plotly_chart(plotly_random_fig(), use_container_width=True)
```

**Comment by Nemecsek** [comment]
No, I didn't. I rewrote the app as a single page, as a temporary solution.

**Comment by jesusprieto-tomtom** [comment]
Hi,

 you should be able to solve this issue by simply adding:

`st.markdown(" <style>iframe{ height: 300px !important } ", unsafe_allow_html=True)`

at the beginning of your main script.

Just keep in mind that this will be affecting to all the iframes in your streamlit app. But in the case your are describing shouldn't be a problem

**Comment by chamath5334** [comment]
@jesusprieto-tomtom 
Can you help me understand -- 
st.markdown(" <style>iframe{ height: 300px !important } ", unsafe_allow_html=True)

...and how it fixes the issue. I have the same problem and it resolved the disappearing eCharts gauge. However, I have a lottie animation right above my eChart that has a height of only 120px and that ends up making that space out too much. Trying to understand this so I might be able to make both work. Thanks in advance.

**Comment by chamath5334** [comment]
@andfanilo Thank you! I take it this is a Streamlit issue since -- 'clicking the tab does not apparently refresh the height of the component'. I opened it up as such and it was closed showing that I should open it as an eCharts and Lottie issue.

**Comment by nicedouble** [comment]
I use **sac.tabs** in my custom components [streamlit-antd-components](https://nicedouble-streamlitantdcomponentsdemo-app-middmy.streamlit.app/) instand of st.tabs.
